### PR TITLE
lib/cmsis_rtos_v1: Check for mutex owner in cmsis

### LINKED ISSUE #9574: tests/cmsis_rtos_v1 - test_mutex_lock_timeout results in Assertion failure on all targets with PR#9569
The test **tests/cmsis_rtos_v1** results in Assertion failure in test_mutex_lock_timeout sub-test case. 

The changes are mainly related to mutex ownership test. It validates the below statement:
**Only thread that obtains a mutex can release it**

**Arch:** any or qemu_x86
**Board:** any or qemu_x86
**Zephyr PR with code changes:** https://github.com/zephyrproject-rtos/zephyr/pull/9569
**Toolchain:**  ZEPHYR_TOOLCHAIN_VARIANT=zephyr
**Shippable failure Link:** https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/21146/1/tests

**Code snippet to reproduce the error:**
```
#define TIMEOUT 500

osMutexDef(Mutex_ownership);

void tThread_entry_mutex_ownership(void const *arg)
{
	osStatus status;

	/* Try to release mutex, obtained by another thread
	 * Only thread that obtains a mutex can release it
	 */
	status = osMutexRelease((osMutexId)arg);
	zassert_true(status == osErrorResource, "Mutex unexpectedly released");
}

osThreadDef(tThread_entry_mutex_ownership, osPriorityNormal, 1, 0);

void test_mutex_ownership(void)
{
	osThreadId id;
	osMutexId mutex_id;
	osStatus status;

	mutex_id = osMutexCreate(osMutex(Mutex_ownership));
	zassert_true(mutex_id != NULL, "Mutex2 creation failed");

	id = osThreadCreate(osThread(tThread_entry_mutex_ownership), mutex_id);
	zassert_true(id != NULL, "Thread creation failed");

	status = osMutexWait(mutex_id, osWaitForever);
	zassert_true(status == osOK, "Mutex wait failure");

	/* wait for spawn thread to take action */
	osDelay(TIMEOUT);

	/* Release the mutex */
	osMutexRelease(mutex_id);

	osMutexDelete(mutex_id);
}
```

**Error Console Log:**
```
***** Booting Zephyr OS v1.13.0-rc1-4-gf80c5cb4a *****
Running test suite test_cmsis_apis
===================================================================
starting test - test_kernel_start
PASS - test_kernel_start
===================================================================
starting test - test_kernel_systick
PASS - test_kernel_systick
===================================================================
starting test - test_thread_apis
PASS - test_thread_apis
===================================================================
starting test - test_thread_prio
PASS - test_thread_prio
===================================================================
starting test - test_thread_instances
PASS - test_thread_instances
===================================================================
starting test - test_timer
oneshot_callback (Timer 1) = 1
periodic_callback (Timer 2) = 1
periodic_callback (Timer 2) = 2
periodic_callback (Timer 2) = 3
periodic_callback (Timer 2) = 4
periodic_callback (Timer 2) = 5
periodic_callback (Timer 2) = 6
PASS - test_timer
===================================================================
starting test - test_mutex
PASS - test_mutex
===================================================================
starting test - test_mutex_lock_timeout
ASSERTION FAIL [mutex->owner == _kernel.current] @ /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/kernel/mutex.c:217:
```

**Steps to reproduce:**
```
cd zephyr/tests/cmsis_rtos_v1/
rm -rf build && mkdir build && cd build
cmake -D BOARD=qemu_x86 ../
make BOARD=qemu_x86 run

Check the console log.
```

### PR DESCRIPTION
**Pull request by rgundi**
Check for the mutex owner in the cmsis layer itself while releasing.
Fixes #9574.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>

### CODE REVIEW

**Review by spoorthik — APPROVED** [review]

File: tests/cmsis_rtos_v1/src/mutex.c [hunk 1]
```diff
@@ -101,6 +101,9 @@ void tThread_entry_lock_timeout(void const *arg)
 	status = osMutexWait((osMutexId)arg, TIMEOUT - 100);
 	zassert_true(status == osErrorTimeoutResource, NULL);
 
+	status = osMutexRelease((osMutexId)arg);
```
Comment (pswarnak) [inline tests/cmsis_rtos_v1/src/mutex.c]: can we please add a comment mentioning, what the code snippet is validating ?
something like "Mutex Ownership check - try to release a mutex, acquired by another thread"

**Review by pswarnak — CHANGES_REQUESTED** [review]

**Review by pswarnak — COMMENTED** [review]
I know the changes are really minor, but just curious to know, if we follow commit rule for lib too, similar to kernel.
That means, Shall we create two different commits for each 1 liner changes in lib and tests OR just one commit is acceptable for minor changes.

File: tests/cmsis_rtos_v1/src/mutex.c (hunk 1 above)
Comment (rgundi) [reply tests/cmsis_rtos_v1/src/mutex.c]: Addressed.

**Comment by rgundi** [comment]
@pswarnak : Addressed your comments.

**Review by pswarnak — APPROVED** [review]

**Pull request merged**